Objects/descrobject.c

```diff
@@ -137,7 +137,7 @@ classmethod_get(PyMethodDescrObject *descr, PyObject *obj, PyObject *type)
 static PyObject *
 method_get(PyMethodDescrObject *descr, PyObject *obj, PyObject *type)
 {
-    PyObject *res;
+    PyObject *res = NULL;
 
     if (descr_check((PyDescrObject *)descr, obj, &res))
```

Comment:
Maybe descr_check() should set res to NULL when it returns 0, whereas having to modify all functions using it.

Reply:
If you modify descr_check() for that, maybe add a comment to explain that it's only set to NULL to prevent compiler warnings.